### ISSUE TITLE
GROUP BY clause doesn't accept column aliases

### ISSUE THREAD

**Issue by TiagoVentosa** [sqlc-dev/sqlc#1363]
### Version

1.11.0

### What happened?

when a query is done with an alias in a select column and there is a  group by clause using the alias, the code generation fails.  This issue seems to be related with https://github.com/kyleconroy/sqlc/pull/1285

### Relevant log output

```shell
column reference "alias" not found
```

### Database schema

```sql
-- Example queries for sqlc
CREATE TABLE authors (
  id   BIGSERIAL PRIMARY KEY,
  name text      NOT NULL,
  bio  text,
  type text      NOT NULL
);
```

### SQL queries

```sql
-- name: countWithCondition :many
SELECT CASE WHEN type = 'DECEASED' then false ELSE true END as alias, count(*) FROM authors
GROUP BY alias;
```

### Configuration

```yaml
{
  "version": "1",
  "packages": [
    {
      "path": "db",
      "engine": "postgresql",
      "schema": "query.sql",
      "queries": "query.sql"
    }
  ]
}
```

### Playground URL

https://play.sqlc.dev/p/3000f866b2534e1ea7cb0d107056a8ff20b8eddf5dbc0c748710c2a905afb98f

### What operating system are you using?

macOS

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by Hworden** [comment]
Not sure what the issue is but I think I narrowed down where to look.

https://github.com/kyleconroy/sqlc/blob/a56a9c1942f6c7ad91da51c71516175c3366ed5a/internal/compiler/output_columns.go#L75

Looks like it throws the error mentioned on the log here 
https://github.com/kyleconroy/sqlc/blob/a56a9c1942f6c7ad91da51c71516175c3366ed5a/internal/compiler/output_columns.go#L538

**Comment by Hworden** [comment]
@TiagoVentosa hey I just built this from source and I don't see the error, however, when I go to version v1.11.0 (commit ad53cfe) I see the error you mentioned. So seems as though between the latest commit and version v1.11.0 the issue was fixed.

I'm attaching the results of sqlc when I build from source using the details you provided.

[results.zip](https://github.com/kyleconroy/sqlc/files/7791557/results.zip)
